pkgs/build-support/php/build-php.nix

```diff
@@ -0,0 +1,106 @@
+{ stdenvNoCC, lib, writeTextDir, composer, php }@inputs:
+
+{ pname
+, version
+, src
+, composerLock ? "__unset"
+, vendorDir ? "vendor"
+, vendorHash
+, php ? inputs.php
+, composer ? inputs.composer
+, ...
+}@args:
+
+let
+  vendor = stdenvNoCC.mkDerivation (finalAttrs: {
+    inherit src version;
+
+    pname = "${pname}-vendor";
+
+    nativeBuildInputs = [
+      composer
+    ];
+
+    strictDeps = true;
+
+    configurePhase = ''
+      runHook preConfigure
+
+      if [ -f ${src}/composer.lock ]
+      then
+        cp -f ${src}/composer.lock composer.lock
+      fi
+
+      if [ -f ${composerLock} ]
+      then
+        cp -f ${composerLock} composer.lock
+      fi
```

Comment:
This will silently do nothing when `composerLock` is given wrong path and `composer.lock` exists in `src`.

Reply:
Fixed by using `elif` insteaf of two successives `if`.

Reply:
Since I moved it in `buildPhase`, the first `if` is now gone.